front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -23,9 +26,14 @@ type MapSearchStationProps = {
 const MapSearchStation = ({ updateExtViewport, closeMapSearchPopUp }: MapSearchStationProps) => {
   const map = useSelector(getMap);
   const [searchState, setSearch] = useState<string>('');
+  const [chCodeFilter, setChCodeFilter] = useState<string>('');
+  const [bvOnly, setBvOnly] = useState<boolean>(false);
   const [searchResults, setSearchResults] = useState<
     SearchResultItemOperationalPoint[] | undefined
   >(undefined);
+  const [filteredSearchResults, setFilteredSearchResults] = useState<
+    SearchResultItemOperationalPoint[] | undefined
+  >(undefined);
```

Comment:
Do we need another state for that ? We could keep using searchResults state and adapt the main useEffect like this : 

```suggestion
useEffect(() => {
    let results = [...trigramResults, ...nameResults];
    if (searchResults) {
      if (bvOnly) {
        results = results.filter((result) => isStation(result.ch));
      } else if (chCodeFilter.trim().length !== 0 && chCodeFilter) {
        results = results.filter((result) => result.ch.startsWith(chCodeFilter.toUpperCase()));
      }
    }
    setSearchResults(results);
  }, [trigramResults, nameResults, bvOnly, chCodeFilter]);
```